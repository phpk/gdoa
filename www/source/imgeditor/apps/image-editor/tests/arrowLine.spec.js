import ArrowLine from '@/extension/arrowLine';

describe('ArrowLine', () => {
  let ctx, arrowLine, linePath;
  function assertPointsToMatchSnapshots() {
    const [firstPoint] = ctx.moveTo.mock.calls;
    const [secondPoint] = ctx.lineTo.mock.calls;
    const [, lastPoint] = ctx.lineTo.mock.calls;

    expect(firstPoint).toMatchSnapshot();
    expect(secondPoint).toMatchSnapshot();
    expect(lastPoint).toMatchSnapshot();
  }

  beforeEach(() => {
    ctx = {
      lineWidth: 1,
      beginPath: jest.fn(),
      moveTo: jest.fn(),
      lineTo: jest.fn(),
      closePath: jest.fn(),
    };
    arrowLine = new ArrowLine();
    arrowLine.ctx = ctx;
    linePath = {
      fromX: 1,
      fromY: 1,
      toX: 10,
      toY: 10,
    };
  });

  it('should draw the "v" calculated according to the angle around the "tail" of the line when attaching the "chevron" type to the end point', () => {
    arrowLine.arrowType = { tail: 'chevron' };
    arrowLine._drawDecoratorPath(linePath);

    assertPointsToMatchSnapshots();
  });

  it('should draw the "v" calculated according to the angle around the "head" of the line when attaching the "chevron" type to the start point', () => {
    arrowLine.arrowType = { head: 'chevron' };
    arrowLine._drawDecoratorPath(linePath);

    assertPointsToMatchSnapshots();
  });

  it('should be a triangular shape that closes the path with closePath after drawing', () => {
    arrowLine.arrowType = { head: 'triangle' };
    arrowLine._drawDecoratorPath(linePath);

    assertPointsToMatchSnapshots();
    expect(ctx.closePath).toBeCalledTimes(1);
  });
});
