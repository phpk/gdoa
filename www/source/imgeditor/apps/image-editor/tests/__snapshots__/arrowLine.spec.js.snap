// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowLine should be a triangular shape that closes the path with closePath after drawing 1`] = `
Array [
  1.4399923766742853,
  2.642073904691416,
]
`;

exports[`ArrowLine should be a triangular shape that closes the path with closePath after drawing 2`] = `
Array [
  1,
  1,
]
`;

exports[`ArrowLine should be a triangular shape that closes the path with closePath after drawing 3`] = `
Array [
  2.642073904691416,
  1.4399923766742853,
]
`;

exports[`ArrowLine should draw the "v" calculated according to the angle around the "head" of the line when attaching the "chevron" type to the start point 1`] = `
Array [
  1.698811421776806,
  3.6079997309804845,
]
`;

exports[`ArrowLine should draw the "v" calculated according to the angle around the "head" of the line when attaching the "chevron" type to the start point 2`] = `
Array [
  1,
  1,
]
`;

exports[`ArrowLine should draw the "v" calculated according to the angle around the "head" of the line when attaching the "chevron" type to the start point 3`] = `
Array [
  3.6079997309804845,
  1.698811421776806,
]
`;

exports[`ArrowLine should draw the "v" calculated according to the angle around the "tail" of the line when attaching the "chevron" type to the end point 1`] = `
Array [
  9.301188578223194,
  7.3920002690195155,
]
`;

exports[`ArrowLine should draw the "v" calculated according to the angle around the "tail" of the line when attaching the "chevron" type to the end point 2`] = `
Array [
  10,
  10,
]
`;

exports[`ArrowLine should draw the "v" calculated according to the angle around the "tail" of the line when attaching the "chevron" type to the end point 3`] = `
Array [
  7.3920002690195155,
  9.301188578223194,
]
`;
